### DISCUSSION TITLE
Register-PnPEntraIDApp Hangs

### DISCUSSION THREAD

**Discussion by KevinWGagel** [pnp/powershell#4407] [Q&A] [answered]
Register-PnPEntraIDApp hangs after providing the warning message. It never completes or provides any kind of error message.
`$result = Register-PnPEntraIDApp -ApplicationName "SrvConSPListUpdate" -Tenant "OURTENANT.onmicrosoft.com" -OutPath "C:\MyFiles\mycertificates" -DeviceLogin
WARNING:
 No permissions specified, using default permissions`

The curser just sits there flashing...

Can someone point me in the right direction?

**Comment by gautamdsheth** [comment, marked as answer]
@KevinWGagel  - this has been fixed in nightly builds. You can use the nightly builds to create the app.

**Reply by KevinWGagel** [reply]
I did find that info in a closed bug and am now using the nightly build.

**Reply by frazeraccount** [reply]
I'm using 2.99.23 nightly and Register-PnPEntraIDApp -ApplicationName "PNPTEST" -Tenant <ourtenantname>.onmicrosoft.com -OutPath C:\test -DeviceLogin hangs after I successfully authenticate and close the popup.

**Reply by erobillard** [reply]
Tested today and the Oct 20 (2.99.18) build worked for me, though I also needed the -SkipPublisherCheck param as this build is NOT signed. @frazeraccount - I was running PS as admin, aside from the slightly earlier build that's the only other possible difference I can think of for you to try.

Note that using a ClientId AND a .PFX certificate is the only way to Connect-PnPOnline to a customer's tenant to use PnP.PowerShell at all (interactive and web login connections are now refused), and particularly to assign permissions to an Azure Automation Account. To get this running in PRD we're going to need a signed release and preferably an "official" release as soon as possible. 

In the meantime, thanks for the fix! 
-Eli.
